packages/storage-ui/src/Contexts/StorageContext.tsx

```diff
@@ -137,53 +157,107 @@ const StorageProvider = ({ children }: StorageContextProps) => {
-  //   const uploadProgress: UploadProgress = {
-  //     id,
-  //     fileName: files[0].name, // TODO: Do we need this?
-  //     complete: false,
-  //     error: false,
-  //     noOfFiles: files.length,
-  //     progress: 0,
-  //     path
-  //   }
-  //   dispatchUploadsInProgress({ type: "add", payload: uploadProgress })
-  //   try {
-  //     // TODO: Make API Request to upload here
+  const createBucket = useCallback((name: string) => {
+    storageApiClient.createBucket({
+      name,
+      type: "fps",
+      public: "read",
+      encryption_key:""
+    }).then(() =>
+      refreshBuckets()
+    ).catch(console.error)
+  }, [storageApiClient, refreshBuckets])
+
+  const removeBucket = useCallback((id: string) => {
+    storageApiClient.removeBucket(id)
+      .then(refreshBuckets)
+      .then(Promise.resolve)
+      .catch(console.error)
+  }, [storageApiClient, refreshBuckets])
 
-  //     // setting complete
-  //     dispatchUploadsInProgress({ type: "complete", payload: { id } })
-  //     setTimeout(() => {
-  //       dispatchUploadsInProgress({ type: "remove", payload: { id } })
-  //     }, REMOVE_UPLOAD_PROGRESS_DELAY)
+  const uploadFiles = useCallback(async (bucketId: string, files: File[], path: string) => {
+    const bucket = storageBuckets.find(b => b.id === bucketId)
 
-  //     return Promise.resolve()
-  //   } catch (error) {
-  //     console.error(error)
-  //     // setting error
-  //     let errorMessage = t`Something went wrong. We couldn't upload your file`
+    if (!bucket) {
+      console.error("Destination bucket does not exist.")
+      return
+    }
 
-  //     // we will need a method to parse server errors
-  //     if (Array.isArray(error) && error[0].message.includes("conflict")) {
-  //       errorMessage = t`A file with the same name already exists`
-  //     }
-  //     dispatchUploadsInProgress({
-  //       type: "error",
-  //       payload: { id, errorMessage }
-  //     })
-  //     setTimeout(() => {
-  //       dispatchUploadsInProgress({ type: "remove", payload: { id } })
-  //     }, REMOVE_UPLOAD_PROGRESS_DELAY)
-  //   }
-  // }, [pins])
+    const id = uuidv4()
+    const uploadProgress: UploadProgress = {
+      id,
+      fileName: files[0].name,
+      complete: false,
+      error: false,
+      noOfFiles: files.length,
+      progress: 0,
+      path
+    }
+    dispatchUploadsInProgress({ type: "add", payload: uploadProgress })
+
+    try {
+      const filesParam = await Promise.all(
+        files
+          .filter((f) => f.size <= MAX_FILE_SIZE)
+          .map(async (f) => {
+            const fileData = await readFileAsync(f)
+            return {
+              data: new Blob([fileData], { type: f.type }),
+              fileName: f.name
+            }
+          })
+      )
+      if (filesParam.length !== files.length) {
+        addToastMessage({
+          message:
+              "We can't encrypt files larger than 2GB. Some items will not be uploaded",
```

Comment:
But there's no ecnryption here rigth?
```suggestion
              "We can't upload files larger than 2GB. Some items will not be uploaded",
```
FMY why this restriction?
Also will this prevent the upload? There's no return here.

Reply:
That's correct, the upload will still happen, but any files larger than 2GB will be excluded.

At the moment, this limit is in place to prevent devices from burning through too much RAM and CPU trying to encrypt large files.